### PR TITLE
Allow to wrap/unwrap collateral

### LINKED ISSUE #187: [FE] - Positions List - Wrap Unwrap buttons + modal
Design: https://zpl.io/2GZznJ9 / https://zpl.io/2GZznJ9 / https://zpl.io/bJyELkn

### LINKED ISSUE #188: [FE] - Position Detail - ERC20 address
See https://github.com/cag/1155-to-20

It is on all networks at this address https://etherscan.io/address/0xec9cc78463b72d7246e8189df5eed5fdc3508e71  the Wrapped1155Factory

### LINKED ISSUE #189: [FE] - Position Detail - Balances box (ERC1155 - ERC20)
See https://github.com/cag/1155-to-20

It is on all networks at this address https://etherscan.io/address/0xec9cc78463b72d7246e8189df5eed5fdc3508e71  the Wrapped1155Factory

### LINKED ISSUE #190: [FE] - Position Detail - Wrap Unwrap buttons + modal

### LINKED ISSUE #194: [FE] - Wrap / Unwrap modals

### LINKED ISSUE #204: [FE] Position List - Balances (ERC20)
See https://github.com/cag/1155-to-20

It is on all networks at this address https://etherscan.io/address/0xec9cc78463b72d7246e8189df5eed5fdc3508e71  the Wrapped1155Factory

### LINKED ISSUE #279: [FE][Bug] ERC1155 sorting on the Positions List table
Sorting by ERC1155 amount seems to be broken:

### LINKED ISSUE #310: Wrapping ERC115 with less than 18 decimals is not working properly
See images

 the wrappedToken just has 18 decimals by default

Screenshot_20200922_171204
Screenshot_20200922_171158

If you fetch the information in thegraph https://thegraph.com/explorer/subgraph/cag/hg-rinkeby?selected=playground
using this query
```
{
  userPositions(where:{position: "0x537d91fcd3aa5753af4cac9cc062923fdbf535c62a10820f0516292ee988e0d2", user:"0x18ad183a875e5a42a60eb5d3a9d6657c3493d064"}) {
    id
    position {
      id
      collateralToken {
        id
      }
      wrappedToken {
        id
      }
    }
    balance
    wrappedBalance
    totalBalance
    user {
      id
    }
  }
}
```

Wrapper contract token generated https://rinkeby.etherscan.io/token/0xae628b5924efd6ab54885e55e09cedfb067a4ce2
See 18 decimals

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #187 
Closes #194 
Closes #190 
Closes #204 
Closes #188 
Closes #189 
Closes #279 
Closes #310

This PR will allow to wrap/unwrapp collateral in position list section, and the position details section.

NOTE: The  transaction work with newly created condition/positions (unless a new update change this behavior)
The PR probably will not work,because we need to put some new env vars to the preview, so check this locally

Also closes the sort action in the position list

### CODE REVIEW

File: scripts/generateGQLTypes.js [hunk 1]
```diff
@@ -2,13 +2,13 @@ const { exec } = require('child_process')
 require('dotenv-flow').config()
 
 exec(
-  `apollo client:codegen src/types/generatedGQL.ts --endpoint ${process.env.REACT_APP_GRAPH_HTTP_MAINNET} --outputFlat --target typescript`,
+  `apollo client:codegen src/types/generatedGQL.ts --endpoint ${process.env.REACT_APP_GRAPH_HTTP_RINKEBY} --outputFlat --target typescript`,
   err => {
```
Comment (mariano-aguero) [inline scripts/generateGQLTypes.js:6]: Changed to rinkeby, because the mainnet subgraph is not updated

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr303--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr303--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — CHANGES_REQUESTED** [review]
Seems to work fine overall. I just found a few issues:

- If I wrap `10` ERC1155 (DAI) everything is fine, I get `10` WMT ERC20. The same goes for most tokens.
- But if I wrap `6.66` ERC1155 (USDC), I get `0.00000000000666` WMT ERC20.
- And if I wrap `3.32` ER1155 ([WBTC](https://rinkeby.etherscan.io/token/0x577d296678535e4903d59a4c929b718e1d575e0a)), I get `0.000000000332` WMT ERC20.

I guess it has something to do with decimals of precision.

File: src/hooks/useCollateral.tsx [hunk 2]
```diff
@@ -16,35 +16,30 @@ export const useCollateral = (
...
-          setError(null)
+    if (collateralAddress) {
+      setLoading(true)
+      const fetchBalanceAndTokenInformation = async () => {
+        try {
+          const erc20Service = new ERC20Service(provider, collateralAddress)
+          const token = await erc20Service.getProfileSummary()
+          if (!cancelled) {
+            setCollateral(token)
+          }
```
Comment (hpmaxi) [inline src/hooks/useCollateral.tsx:27]: loading and error never changes.

**Review by hpmaxi — APPROVED** [review]

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr303--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr303--conditionaltokensexplorer.review.gnosisdev.com

**Review by gabitoesmiapodo — APPROVED** [review]

**Pull request merged**